boards/arm/stm32/nucleo-f302r8/configs/qenco/defconfig

```diff
@@ -17,24 +71,40 @@ CONFIG_ARCH_INTERRUPTSTACK=1024
 CONFIG_ARCH_IRQPRIO=y
 CONFIG_BOARD_LOOPSPERMSEC=8499
 CONFIG_BUILTIN=y
-CONFIG_DEBUG_FULLOPT=y
+CONFIG_DEBUG_CUSTOMOPT=y
+CONFIG_DEBUG_OPTLEVEL="-Os"
 CONFIG_DEBUG_SYMBOLS=y
+CONFIG_DEFAULT_SMALL=y
 CONFIG_EXAMPLES_QENCODER=y
 CONFIG_EXAMPLES_QENCODER_HAVE_MAXPOS=y
 CONFIG_EXAMPLES_QENCODER_MAXPOS=8192
+CONFIG_FILE_STREAM=y
 CONFIG_INIT_ENTRYPOINT="nsh_main"
 CONFIG_INTELHEX_BINARY=y
+CONFIG_LIBC_FLOATINGPOINT=y
+CONFIG_LIBC_LONG_LONG=y
+CONFIG_LIBM_TOOLCHAIN=y
 CONFIG_MQ_MAXMSGSIZE=5
+CONFIG_NSH_ALIAS=y
 CONFIG_NSH_ARCHINIT=y
+CONFIG_NSH_ARGCAT=y
 CONFIG_NSH_BUILTIN_APPS=y
-CONFIG_PREALLOC_TIMERS=4
+CONFIG_NSH_CLE=y
+CONFIG_NSH_CMDOPT_HEXDUMP=y
+CONFIG_NSH_FILEIOSIZE=1024
+CONFIG_NSH_LINELEN=80
+CONFIG_NSH_QUOTE=y
```

Comment:
I think that this also can be set to `n`

Reply:
I'll test disabling these configs but as I edited my commented above, I think the only way will be to disable the NSH.

Reply:
Maybe it is enough. No error was thrown
```
lucassvaz@Lucas-Espressif:~/nuttxspace/nuttx$ make -j
Create version.h
LN: platform/board to /home/lucassvaz/nuttxspace/apps/platform/dummy
Register: qe
Register: nsh
Register: sh
LD: nuttx
CP: nuttx.hex
CP: nuttx.bin
lucassvaz@Lucas-Espressif:~/nuttxspace/nuttx$ size nuttx
   text    data     bss     dec     hex filename
  64176     332    3252   67760   108b0 nuttx
```

Reply:
Yes. Disable NSH is a good option. As well as disabling everything that is not used by application on the tiny platforms. Like if apps do not use message queues then those should be disabled. The same with POSIX timers, etc. In general I think most of those will be removed by -ffunctio-sections, but need to analyze map files to see that truth.